prow/cmd/validation-webhook/main.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+	"k8s.io/test-infra/experiment/clustersecretbackup/secretmanager"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+const (
+	secretID  = "ca-certification"
+	secretMapKey = "ca-secrets"
+	caCert = "ca-cert"
+	caPrivKey = "ca-priv-key"
+	caBundle = "ca-bundle"
+	latest = "projects/*/secrets/*/versions/latest"
+	gcpSecretError = "could not get secret data"
+)
+
+type ClientInterface interface {
+	CreateSecret(ctx context.Context, secretID string) (*secretmanagerpb.Secret, error)
+	AddSecretVersion(ctx context.Context, secretName string, payload []byte) error
+	GetSecretValue(ctx context.Context, secretName, versionName string) ([]byte, error)
+}
+
+
+
+func main() {
+	logrusutil.ComponentInit()
+	logrus.SetLevel(logrus.DebugLevel)
+	// GCP project in which to store secrets in Secret Manager
+	dnsNames := []string{"validation-webhook-service",
+		"validation-webhook-service.default", "validation-webhook-service.default.svc"}
+	p := flag.String("p", 
+	"colew-test", "Project ID for storing GCP Secrets")
+	e := flag.Int("expiry-years", 30, "CA certificate expiry in years")
+	dns := flagutil.NewStrings()
+	for _, dnsName := range(dnsNames) {
+		dns.Add(dnsName)
+	}
+	flag.Var(&dns, "dns", "DNS Names CA-Cert config")
+	flag.Parse()
+	exp := *e
+	projectId := *p
+	client, err := secretmanager.NewClient(projectId, false)
+	if err != nil {
+		logrus.WithError(err).Fatal("Unable to create secret manager client")
+	}
+	ctx := context.Background()
+	cert, privKey, err := getGCPSecrets(client, ctx, exp, dns)
+	if err != nil && err.Error() == gcpSecretError {
+		logrus.WithError(err)
```

Comment:
This doesn't do anything, it returns a log entry that isn't used.

Reply:
yeah im just letting the user know that it could not get secret data, before trying to create the secret

Reply:
This seems to be intended behavior. If there is an error here we attempt to create the secret? If so, we do not want to spam the logs with errors that we expect to hit. Maybe make this debug or info with more information that we are attempting to create the secret.